improver/calibration/rainforest_calibration.py

```diff
@@ -395,64 +431,43 @@ def _make_decreasing(self, probability_data: ndarray) -> ndarray:
         return 0.5 * (upper + lower)
 
     def _evaluate_probabilities(
-        self,
-        forecast_data: ndarray,
-        input_data: ndarray,
-        forecast_variable: str,
-        forecast_variable_unit: str,
-        output_data: ndarray,
+        self, input_data: ndarray, lead_time_hours: int, output_data: ndarray,
     ) -> None:
-        """Evaluate probability that error in forecast exceeds thresholds, setting
-        the result to 1 when `forecast + threshold` is less than or equal to
-        the lower bound of forecast_variable, as defined in constants.BOUNDS_FOR_ECDF`.
+        """Evaluate probability that forecast exceeds thresholds.
 
         Args:
-            forecast_data:
-                1-d containing data for the variable to be calibrated.
             input_data:
                 2-d array of data for the feature variables of the model
-            forecast_variable:
-                name of forecast variable
-            forecast_variable_unit:
-                unit of forecast variable
+            lead_time_hours:
+                lead time in hours
             output_data:
                 array to populate with output; will be modified in place
         """
 
         input_dataset = self.model_input_converter(input_data)
 
-        bounds_data = get_bounds_of_distribution(
-            forecast_variable, forecast_variable_unit
-        )
-        lower_bound_in_fcst_units = bounds_data[0]
-
-        for threshold_index, model in enumerate(self.tree_models):
-            # it is assumed that models have been trained using the >= operator
-            # (i.e. they predict the probabilty that error >= threshold)
-            threshold = self.error_thresholds[threshold_index]
-            if threshold >= 0:
-                # In this case, for all values of forecast we have
-                # forecast + threshold >= forecast >= lower_bound_in_fcst_units
-                prediction = model.predict(input_dataset)
-            else:
-                # In this case, we have error > threshold if and only if
-                # observations > forecast + threshold, which has probability 1
-                # if forecast + threshold < lower_bound_in_fcst_units
-                prediction = np.ones(input_data.shape[0], dtype=np.float32)
-                forecast_bool = forecast_data + threshold >= lower_bound_in_fcst_units
-                if np.any(forecast_bool):
-                    input_subset = self.model_input_converter(input_data[forecast_bool])
-                    prediction[forecast_bool] = model.predict(input_subset)
+        if np.float32(lead_time_hours) in self.lead_times:
```

Comment:
I've changed the following line to be `model_lead_time = np.float32(lead_time_hours)`. We can't use `np.isclose` for the comparison, because in the case that `np.float32(lead_time_hours)` does not exactly match one of `self.lead_times`, the subsequent lookup (`model = self.tree_models[model_lead_time, threshold]`) will fail.